src/shared/Microsoft.Git.CredentialManager/Settings.cs

```diff
@@ -392,6 +398,30 @@ public bool IsCertificateVerificationEnabled
             }
         }
 
+        public string SslCaInfo
+        {
+            get
+            {
+                // First, verify that, if schannel is the HTTP backend, sslCaInfo must be explicitly enabled to be used
+                if (TryGetSetting(null, KnownGitCfg.Http.SectionName, KnownGitCfg.Http.SslBackend, out string backend) &&
+                    StringComparer.OrdinalIgnoreCase.Equals(backend, "schannel") &&
+                    (!TryGetSetting(null, KnownGitCfg.Http.SectionName, KnownGitCfg.Http.SchannelUseSslCaInfo, out string schannelUseSslCaInfo) ||
+                    !schannelUseSslCaInfo.ToBooleanyOrDefault(false)))
+                {
+                    return null;
+                }
```

Comment:
Personally I would say the `SslCaInfo` property on `ISettings` should return the _actual_ resolved value of `GIT_SSL_CAINFO/http.sslCAInfo`, and _the caller_ should be responsible for enforcing the "only if the backend is not schannel then use the value".

Would mean exposing both the `sslBackend` and `schannelUseSSLCAInfo` settings as additional properties on `ISettings`, which is probably a good idea.

Reply:
I can think of 3 potential ways to handle the setting:

1. What you're suggesting - the caller handles compatibility checks with `schannel`
2. Create/rename to a "new" setting like `CustomSslCertificateFile` (similar `IsCertificateVerificationEnabled`, which isn't quite the same as `GIT_SSL_NO_VERIFY`) that internally handles the `schannel` compatibility
3. Instead of a property (as in 2), create a method like `GetCustomSslCertificateFile()` (similar to `GetProxyConfiguration()`)

I like the testability of 2 & 3 (and that they represent more of a "black box" setting that is guaranteed to match Git behavior without any extra checks in potentially multiple places), but if there are scenarios where you'd want the directly-from-config value, then 1 would probably be the best way to go.

Reply:
I think the responsibility of applying the cert validation rules lies with the `HttpClientFactory` (since that's where the validation callback also lives). The fact that sslNoVerify wins out to the sslCAInfo is also encoded in how the factory evaluates the settings.

You are right of course that `IsCertificateVerificationEnabled` is not exactly like NoVerify (it's a boolean `NOT` of the underlying setting). I guess that should have been called `IsCertificateVerificationIgnored` really.

The more complex proxy settings implementation is because we're trying to hide somewhat the back-compat of now deprecated proxy setting mechanisms, but also want to warn the user about the specific values mechanisms being used.

I think I'd still prefer option 1, with the following properties, and possibly an `enum` for the backend type:

```csharp
// With this enum we can avoid having to create externally visible constants for "schannel" etc
enum TlsBackend
{
    OpenSsl,
    Schannel,
}

interface ISettings
{
    TlsBackend TlsBackend { get; }
    string CustomRootCertificatePath { get; }
    bool UseCustomRootCertificateWithSchannel { get; }
}
```

Really it's TLS these days, so let's stick with "TLS" in the interface and up.

Reply:
This should be updated now! But please double-check that what's there now approximately matches what you were thinking (especially the `TlsBackend` enum).